source/main.cc

```diff
@@ -450,6 +450,7 @@ int main (int argc, char *argv[])
 #ifdef ASPECT_USE_FP_EXCEPTIONS
   // enable floating point exceptions
   feenableexcept(FE_DIVBYZERO|FE_INVALID);
+  feclearexcept(FE_DIVBYZERO|FE_INVALID);
```

Comment:
That is a misunderstanding. The bits indicating that an exception happened are always set whenever a floating-point exception happens. `feenableexcept` only enables to throw an exception if the bits are set. The opposite function would be `fedisableexcept`.
`feclearexcept` only makes sure these bits are not set at the beginning of our program call. I could reverse the order of the commands and would get the same effect.
Usually these bits should be initialized to zero, but they are inherited from the parent thread.  My guess is that these bits are already set on some clusters, because our executable is started by some launcher (e.g. `aprun` or `ibrun`). FPE's inside of the launcher might have already set the bits, but might not throw the exceptions, or caught them and handled them already. This would also explain why our cmake test passes, because in that case the executable was not started by the launcher, so the bits are not already set.

I tested that with the changes of this PR inserting the following lines right after the block still throw a FE_DIVBYZERO FPE:
```
double inf = 1.0 / 0.0;
std::cout << inf;
```